src/zenml/integrations/mlflow/experiment_trackers/mlflow_experiment_tracker.py

```diff
@@ -219,34 +220,37 @@ def get_step_run_metadata(
         }
 
     def disable_autologging(self) -> None:
-        """Disables MLflow autologging."""
-        from mlflow import (
-            fastai,
-            gluon,
-            lightgbm,
-            pytorch,
-            sklearn,
-            spark,
-            statsmodels,
-            tensorflow,
-            xgboost,
-        )
-
-        # There is no way to disable auto-logging for all frameworks at once.
-        # If auto-logging is explicitly enabled for a framework by calling its
-        # autolog() method, it cannot be disabled by calling
-        # `mlflow.autolog(disable=True)`. Therefore, we need to disable
-        # auto-logging for all frameworks explicitly.
-
-        tensorflow.autolog(disable=True)
-        gluon.autolog(disable=True)
-        xgboost.autolog(disable=True)
-        lightgbm.autolog(disable=True)
-        statsmodels.autolog(disable=True)
-        spark.autolog(disable=True)
-        sklearn.autolog(disable=True)
-        fastai.autolog(disable=True)
-        pytorch.autolog(disable=True)
+        """Disables MLflow autologging for all supported frameworks."""
+        frameworks = [
+            "tensorflow",
+            "gluon",
+            "xgboost",
+            "lightgbm",
+            "statsmodels",
+            "spark",
+            "sklearn",
+            "fastai",
+            "pytorch",
+        ]
+
+        failed_frameworks = []
+
+        for framework in frameworks:
+            try:
+                # Correctly prefix the module name with 'mlflow.'
+                module_name = f"mlflow.{framework}"
+                # Dynamically import the module corresponding to the framework
+                module = importlib.import_module(module_name)
+                # Call the autolog function with disable=True
+                module.autolog(disable=True)
+            except Exception:
```

Comment:
Ah but in the case of an `ImportError` its not really failing to disable autologging , its just that the module was not installed right? :-) So id bring back the ImportError and do a `pass` in that block instead. Rest LGTM!

Reply:
I'm a bit confused - because the module is required to disable autologging.

If the import fails at `:243` (raising the `ImportError`), it will never go to disabling autologging at `:245` and hence it failed to disable autologging.

Reply:
@avishniakov if the import fails, it cannot disable autologging, so disabling autologging failed, right? Then, why omit adding the framework to `failed_frameworks` on failed autologging?

Reply:
In case of _any_ error (whether that's during import or during autologging), the MLFlow module failed to disable autologging. In _any_ case, it should then be added to `failed_frameworks`, right? What am I missing in my line of thinking that would warrant extra code distinguishing an `ImportError` from any other `Exception`?

Reply:
I think for me the distinguishing fact is that `ImportError` is a soft error that  can happen and is expected becasue nobody has all the modules installed usually, but the other errors are things that are more noticable and should notify the user

Reply:
Sure, I understand, but if `ImportError` is raised (because for example, in the original case, `tensorflow` is not available), then from my understanding the suggestion below will never disable autologging, and will **neither** inform the user about this because it is not passed into `failed_frameworks`. 

See this example:

```
failed = []

try:
    raise ImportError("Failed tensorflow import")
except ImportError:
    pass
else:
    failed.append("tensorflow")

print(failed)
assert len(failed) == 1 and failed[0] == "tensorflow"
```

```
(base) C:\Users\cvers\Documents>python zenmltest.py
[]
Traceback (most recent call last):
  File "C:\Users\cvers\Documents\zenmltest.py", line 11, in <module>
    assert len(failed) == 1 and failed[0] == "tensorflow"
```

Translating this to the example below using the TF case:

* If within `mlflow.tensorflow` TensorFlow itself fails to import, disable autolog is never called
* This means that the MLFlow experiment will continue autologging
* This leads to vague behavior from a user perspective, as something orthogonal to what the user expects is happening.

If you'd argue that the user will need `tensorflow` when using the MLFlow TensorFlow module and hence will always have it when requiring disabling autologging, I can see your point and would agree. Otherwise, I'd like to understand it better.

Reply:
> If you'd argue that the user will need tensorflow when using the MLFlow TensorFlow module and hence will always have it when requiring disabling autologging, I can see your point and would agree. Otherwise, I'd like to understand it better.

yes, this is my point but i could be wrong. @avishniakov WDYT?

Reply:
@htahir1 @christianversloot  No strong arguments on my end, but I would love to see logging more explicit if we combine it - catch `ImportError` and `Exception` separately and add a clear message to the user that the library is missing or there is some other unexpected error coming from MLFlow.

In my opinion, if import failed execution anyway will fail somewhere down the road or most probably failed even before on same import in step file, so I would not be worried about this `ImportError` thingy and guess that this one could be merged once CI is green.